controllers/transactions.js

```diff
@@ -0,0 +1,118 @@
+const Transaction = require('../models/Transaction');
+
+// @desc:    Get all transactions
+// @route:   GET /api/v1/transactions
+// @access:  Private
+exports.getTransactions = async (req, res, next) => {
+  let query;
+  console.log(req);
+
+  const reqQuery = {...req.query};
+  const removeFields = ["select", "sort", "page", "limit"];
+
+  // Loop over remove fields and delete from reqQuery
+  removeFields.forEach((params) => delete reqQuery[params]);
+
+  let queryStr = JSON.stringify(req.query);
+
+  // Create operators ($gt, $gte, etc.)
+  queryStr = queryStr.replace(
+    /\b(gt|gte|lt|lte|in)\b/g,
+    (match) => `$${match}`
+  );
+
+  // Select
+  query = Transaction.find(JSON.parse(queryStr));
+
+  // Sort
+  if (req.query.select) {
+    const sortBy = req.query.sort.split(",").join(" ");
+    query = query.sort(sortBy);
+  } else {
+    query = query.sort("name");
+  }
+
+  // Pagination
+  const page = parseInt(req.query.page, 10) || 1;
+  const limit = parseInt(req.query.limit, 10) || 10;
+  const startIndex = (page - 1) * limit;
+  const endIndex = page * limit;
+  const total = await Transaction.countDocuments();
+
+  query = query.skip(startIndex).limit(limit);
+
+  if (req.user.role !== "admin") {
```

Comment:
from routes/transactions.js only the admins are allowed to access this API. are user able to access this API too?